jobs/buildhub/to_kinto.py

```diff
@@ -148,13 +234,11 @@ def done(future):
             return results
         return done
 
-    def records_equal(r1, r2):
-        omit = ['last_modified', 'schema']
-        r1c = {k: v for k, v in r1.items() if k not in omit}
-        r2c = {k: v for k, v in r2.items() if k not in omit}
-        return r1c == r2c
-
-    records_by_id = {r['id']: r for r in existing}
+    def record_unchanged(record):
+        return (
+            record['id'] in existing and
+            existing.get(record['id']) == hash_record(record)
```

Comment:
Since `hash_record()` never returns `None`, you can write:
```py
return existing.get(record['id']) == hash_record(record)
```
and thus even maybe drop the use of this `record_unchanged()` function

Reply:
No, if I do that I have to execute `hash_record(record)` just so that it can ultimately become: `if None == 'somehashstringhere':`.

The first part of the if helps figure out if it's worth bothering calculating a hash. By doing both things the if statement becomes (if you break it down in steps): `if False and existing.get(record['id']) == hash_record(record):` which means it exists out on the first `False` and doesn't bother executing the `existing.get(record['id']) == hash_record(record)` part at all. 

Right?

Reply:
yes